app/src/main/java/com/asfoundation/wallet/billing/paypal/PayPalIABViewModel.kt

```diff
@@ -54,6 +54,20 @@ class PayPalIABViewModel @Inject constructor(
   val networkScheduler = rxSchedulers.io
   val viewScheduler = rxSchedulers.main
 
+  fun startPayment(
+    createTokenIfNeeded: Boolean = true, amount: BigDecimal, currency: String,
+    transactionBuilder: TransactionBuilder, origin: String?
+  ) {
+    sendPaymentConfirmationEvent(transactionBuilder)
+    attemptTransaction(
+      createTokenIfNeeded = createTokenIfNeeded,
+      amount = amount,
+      currency = currency,
+      transactionBuilder = transactionBuilder,
+      origin = origin
+    )
+  }
+
   fun attemptTransaction(
```

Comment:
why don’t use this old function and call the event here, it looks duplicated

Reply:
It's necessary because the event should only be sent one time when starting the payment. This attemptTransaction method is called twice for paypal